src/array/to_geoarrow.rs

```diff
@@ -69,49 +74,23 @@ pub trait ToWKBPolygons {
     ) -> Result<WKBArray<O>, Self::Error>;
 }
 
-impl<T> ToWKBPolygons for T
+impl<T> ToWKBPolygons for H3Array<T>
 where
-    T: ToPolygons,
+    Self: IterPolygons,
+    T: H3IndexArrayValue,
 {
-    type Error = T::Error;
+    type Error = <Self as IterPolygons>::Error;
 
     fn to_wkb_polygons<O: OffsetSizeTrait>(
         &self,
         use_degrees: bool,
     ) -> Result<WKBArray<O>, Self::Error> {
-        Ok(WKBArray::from(
-            self.to_polygons(use_degrees)?
-                .into_iter()
-                .map(|v| v.map(Geometry::from))
-                .collect::<Vec<_>>(),
-        ))
-    }
-}
-
-pub trait ToWKBLines {
-    type Error;
-    fn to_wkb_lines<O: OffsetSizeTrait>(
-        &self,
-        use_degrees: bool,
-    ) -> Result<WKBArray<O>, Self::Error>;
-}
-
-impl<T> ToWKBLines for T
-where
-    T: ToLines,
-{
-    type Error = T::Error;
-
-    fn to_wkb_lines<O: OffsetSizeTrait>(
-        &self,
-        use_degrees: bool,
-    ) -> Result<WKBArray<O>, Self::Error> {
-        Ok(WKBArray::from(
-            self.to_lines(use_degrees)?
-                .into_iter()
-                .map(|v| v.map(Geometry::from))
-                .collect::<Vec<_>>(),
-        ))
+        let mut builder = WKBBuilder::with_capacity(WKBCapacity::new(self.len(), self.len()));
```

Comment:
btw, depending on the relative cost of repeated allocations to grow the wkb array vs the cost of materializing geometries from h3 objects, you may want to pre-calculate the exact capacities that you need for the `WKBArray`.

You can do that with [`WKBCapacity`](https://docs.rs/geoarrow/latest/geoarrow/array/struct.WKBCapacity.html) by calling [`add_polygon`](https://docs.rs/geoarrow/latest/geoarrow/array/struct.WKBCapacity.html#method.add_polygon)

Reply:
Thant you for your review - that is a good point. I tried to address this with 4dcf0caf49538ae4367eb7c839264eee67b288ce